lib/c-api/src/wasm_c_api/types/export.rs

```diff
@@ -42,19 +42,7 @@ impl From<ExportType> for wasm_exporttype_t {
 
 impl From<&ExportType> for wasm_exporttype_t {
     fn from(other: &ExportType) -> Self {
-        let name = {
-            let mut heap_str: Box<str> = other.name().to_string().into_boxed_str();
-            let char_ptr = heap_str.as_mut_ptr();
-            let str_len = heap_str.bytes().len();
-            let name_inner = wasm_name_t {
-                size: str_len,
-                data: char_ptr,
-            };
-            Box::leak(heap_str);
-
-            Box::new(name_inner)
-        };
-
+        let name = Box::new(other.name().to_string().into());
```

Comment:
For readability it'd be nice to explicitly annotate the type here.  I know that rust analyzer and the IDE tools can show it when editing the code, but it's still nice to make the code clear as plain text. Otherwise, I like the code reuse!

Reply:
:+1:

Reply:
Done.